### PR TITLE
Update `birthTime` for validation & add direct xml comparison in test

### LINKED ISSUE #1278: Change and move `<birthTime>` element
## Action Requested
The `<birthTime>` element is both in the wrong place and not formatted correctly. So we'll need to do two things:
1. Move `<birthTime>` directly after `<administrativeGenderCode>`; and,
2. Add a `value` attribute to `<birthTime>` element
  a. change the datatype of the date to string and ensure there are neither spaces nor dashes, e.g., `"19500101"`

If we try and use the expected ISO standard the validator will throw the following error:

```bash
Element '{urn:hl7-org:v3}birthTime', attribute 'value':  
The value '1950-01-01' is not accepted by the pattern 
'[0-9]{1,8}|([0-9]{9,14}|[0-9]{14,14}\.[0-9]+)([+\-][0-9]{1,4})?'.
```

Here's an example of how it should look and where it should be situated:

```xml
        <administrativeGenderCode code="M" codeSystem="2.16.840.1.113883.12.1"
          codeSystemName="Administrative sex (HL7)" displayName="Male">
          <translation code="M" codeSystem="2.16.840.1.113883.12.1"
            codeSystemName="2.16.840.1.113883.12.1" displayName="Male"></translation>
        </administrativeGenderCode>
        <birthTime value="19500101"></birthTime>
        <maritalStatusCode code="S" codeSystem="2.16.840.1.113883.5.2"
          codeSystemName="MaritalStatus" displayName="Never Married">
          <translation code="S" codeSystem="2.16.840.1.113883.5.2"
            codeSystemName="2.16.840.1.113883.5.2" displayName="Single, never married"></translation>
        </maritalStatusCode>
```
## Acceptance Criteria
When the PHDC builder is able to produce XML following the above format, the XML assets have been updated, and the accompanying tests are passing based on the updated input of expected results and what the builder is producing.

## Additional Information
Can use the [validator](https://github.com/robertmitchellv/phdc-validation/) I made to test things on your own as we figure out if we'll add something like this to the message parser or not; and if so, in what way.

### PR DESCRIPTION
**Pull request by m-goggins**
# PULL REQUEST

## Summary
This PR updates the `birthTime` element so that it is in the correct place and formatted correctly with a `value` for validation. I also implemented a new function in utils.py to parse XML files to make it easier to make comparisons of XML trees in our tests (instead of stringified versions of the XMLs). I used [xmldiff](https://xmldiff.readthedocs.io/en/stable/index.html) because it had documentation that was easy to follow and would allow us to compare files, trees, or text versions, but I'm not strongly attached to it if we go another route for testing.

## Related Issue
Fixes #1278 

## Additional Information
Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: containers/message-parser/app/utils.py [hunk 1]
```diff
@@ -327,6 +328,24 @@ def read_file_from_assets(filename: str) -> str:
         return file.read()
 
 
+def parse_file_from_assets(filename: str) -> ET.ElementTree:
```
Comment (DanPaseltiner) [inline containers/message-parser/app/utils.py:331]: Nice!

File: containers/message-parser/app/phdc/builder.py [hunk 2]
```diff
@@ -717,6 +717,17 @@ def _build_patient(self, patient: Patient) -> ET.Element:
             )
             patient_data.append(v)
 
+        if patient.birth_time is not None:
+            e = ET.Element(
+                "birthTime",
+                {
+                    "value": "".join(
+                        [num for num in patient.birth_time if num.isnumeric()]
```
Comment (DanPaseltiner) [inline containers/message-parser/app/phdc/builder.py:724]: This seems a little bit brittle to me, but given the constraints on dates in FHIR I think okay for now at least.

**Review by DanPaseltiner — APPROVED** [review]
LGTM.

File: containers/message-parser/app/utils.py (hunk 1 above)
Comment (robertmitchellv) [reply containers/message-parser/app/utils.py:331]: great addition!

File: containers/message-parser/tests/test_phdc.py [hunk 3]
```diff
@@ -309,27 +311,33 @@ def test_build_author(family_name, expected_oid, expected_date, expected_author)
...
-    assert ET.tostring(xml_patient_data).decode() == expected_result
+    formatter = formatting.DiffFormatter(
+        normalize=formatting.WS_BOTH, pretty_print=True
+    )
+    diff = xmldiff.diff_trees(xml_patient_data, expected_result, formatter=formatter)
+
+    assert diff == ""
+
+
+def elements_equal(e1, e2):
```
Comment (robertmitchellv) [inline containers/message-parser/tests/test_phdc.py:329]: this is great 🚀

**Review by robertmitchellv — APPROVED** [review]
i like the new tests for the xml elements!

**Pull request merged**